src/utilities/typeComparators.js

```diff
@@ -0,0 +1,85 @@
+/* @flow */
+/**
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import {
+  isAbstractType,
+  GraphQLObjectType,
+  GraphQLList,
+  GraphQLNonNull,
+} from '../type/definition';
+import type { GraphQLType, GraphQLAbstractType } from '../type/definition';
+
+
+/**
+ * Provided two types, return true if the types are equal (invariant).
+ */
+export function isEqualType(typeA: GraphQLType, typeB: GraphQLType): boolean {
+  // Equivalent types are equal.
+  if (typeA === typeB) {
+    return true;
+  }
+
+  // If either type is non-null, the other must also be non-null.
+  if (typeA instanceof GraphQLNonNull && typeB instanceof GraphQLNonNull) {
+    return isEqualType(typeA.ofType, typeB.ofType);
+  }
+
+  // If either type is a list, the other must also be a list.
+  if (typeA instanceof GraphQLList && typeB instanceof GraphQLList) {
+    return isEqualType(typeA.ofType, typeB.ofType);
+  }
+
+  // Otherwise the types are not equal.
+  return false;
+}
+
+/**
+ * Provided a parent type and a child type, return true if the second is either
+ * equal or a subset of the second (covariant).
+ */
+export function isSubType(parent: GraphQLType, child: GraphQLType): boolean {
```

Comment:
Hm, I wish there were some way to indicate the ordering here;

```js
isSubType(a, b)
```

doesn't make it immediately clear to me whether we're checking `a` as a subtype of `b` or vice versa.